### PR TITLE
[serverless] Move integration tests into agent repo

### PR DESCRIPTION
**Pull request by nhinsch**
### What does this PR do?
This PR moves the Lambda Extension/Serverless Agent integration tests, which used to be in the separate `datadog-lambda-extension` repository, into the `datadog-agent` repository, so that the tests and their snapshot data are co-located in the same repository as the code that they test. **The tests must be run locally for now, this PR does not add or modify any CI jobs**.

It renames the existing "serverless" integration test that is used to test cold start performance to "serverless-perf".

### Motivation

Having the tests and their snapshot data in a separate repository from the code has proved highly problematic. When we update the Lambda Extension, the tests naturally begin to fail, but we must currently run and update the tests in a separate repository. The tests and the code they are testing are constantly getting out of sync. It is also difficult to see if a given agent PR is breaking the tests and why.

### Additional Notes
**These tests will not run in CI yet.** These tests must be run locally for now. No additional GitHub action or GitLab job has been added at this point. I will try to add one in another PR, perhaps running only on Serverless-related PRs, but this should not block the initial move of the test suite.

### Reviewer's Checklist
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.

### CODE REVIEW

**Review by L3n41c — APPROVED** [review]

**Review by mx-psi — CHANGES_REQUESTED** [review]
You need to address the Python linter errors

File: test/integration/serverless/recorder-extension/go.mod [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module datadog-lambda-extension/recorder-extension
+
+go 1.14
```
Comment (mx-psi) [inline test/integration/serverless/recorder-extension/go.mod:3]: Go 1.14 is no longer supported, ideally you would use Go 1.16 which is what we have installed in our builders

File: test/integration/serverless/recorder-extension/go.mod [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.20.5
+	k8s.io/metrics => k8s.io/metrics v0.20.5
+	k8s.io/mount-utils => k8s.io/mount-utils v0.20.3-rc.0
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.20.5
+	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.20.5
+	k8s.io/sample-controller => k8s.io/sample-controller v0.20.5
+)
+
+require (
+	github.com/DataDog/agent-payload v4.73.0+incompatible // indirect
```
Comment (mx-psi) [inline test/integration/serverless/recorder-extension/go.mod:36]: This should not have `// indirect` on it

File: test/integration/serverless/recorder-extension/go.mod [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	github.com/DataDog/datadog-agent v0.0.0-20210616155722-298368b28184 // indirect
```
Comment (mx-psi) [inline test/integration/serverless/recorder-extension/go.mod:37]: Here you are pinning a specific commit of the Agent, is that okay? Maybe you want to add a local `replace` and use the code in the repo?

Comment (nhinsch) [reply test/integration/serverless/recorder-extension/go.mod]: I copypasted this test suite from another repo, so I'm not 100% sure how everything works here. However, I believe the intent here is to use a pinned version of the `datadog-agent/pkg/trace/pb` module, not to use the local agent code in the repo. I left this as-is for now.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**